OpenRA.Mods.Common/Traits/World/SmudgeLayer.cs

```diff
@@ -163,14 +163,29 @@ public void AddSmudge(CPos loc)
 			}
 		}
 
+		public void RemoveSmudge(CPos loc)
+		{
+			if (dirty.ContainsKey(loc))
+			{
+				var tile = dirty[loc];
+				tile.Sprite = null;
+				dirty[loc] = tile;
+			} else {
+				dirty[loc] = new Smudge { Type = "", Depth = 0, Sprite = null };
```

Comment:
I think you will need to set a valid Type here (so the original random would have been fine), otherwise you will cause a crash if something else calls AddSmudge on the same cell before the dirty cells are resolved.